src/rebar_git_resource.erl

```diff
@@ -253,16 +253,7 @@ git_vsn_fetch() ->
 
 make_vsn(AppInfo, _) ->
     Dir = rebar_app_info:dir(AppInfo),
-    case rebar_app_info:original_vsn(AppInfo) of
```

Comment:
Not entirely sure about this one, as mentioned in https://github.com/erlang/rebar3/pull/2412#issuecomment-718277334.

Reply:
The value we get from here might be much broader than what the typespec specifies by virtue of being consulted there:
https://github.com/erlang/rebar3/blob/5dab01986786f2ffc4ea2b8d34a94177f6f40808/src/rebar_app_discover.erl#L367-L375

The code only shows up as dead because the original_vsn type spec is more conservative than what we see used in the real world. This code is not dead, Dialyzer is mislead due to bad annotations.

Reply:
Should we consider "disabling" `dialyzer` for this? Or find a more generic spec for it?

Reply:
More generic spec would be good. This used to be a more central place where we handled things for example: https://github.com/erlang/rebar3/blob/bed5a1a505981bf0600d6562328739db4c9918ac/src/rebar_utils.erl#L761-L787

We could expect: {cmd, string()}, {file, filename()}, {ResourceProvider::atom(), term()} along with the regular string-based formats.

Reply:
24e02eb